core/server/src/main/java/alluxio/web/UIWebServer.java

```diff
@@ -65,29 +69,36 @@ public UIWebServer(ServiceType service, InetSocketAddress address) {
     QueuedThreadPool threadPool = new QueuedThreadPool();
     int webThreadCount = Configuration.getInt(PropertyKey.WEB_THREADS);
 
-    mServer = new Server();
-    SelectChannelConnector connector = new SelectChannelConnector();
-    connector.setHost(address.getHostName());
-    connector.setPort(address.getPort());
-    connector.setAcceptors(webThreadCount);
-    mServer.setConnectors(new Connector[] {connector});
-
-    try {
-      mServer.getConnectors()[0].close();
-      mServer.getConnectors()[0].open();
-    } catch (IOException e) {
-      throw Throwables.propagate(e);
-    }
-
     // Jetty needs at least (1 + selectors + acceptors) threads.
     threadPool.setMinThreads(webThreadCount * 2 + 1);
     threadPool.setMaxThreads(webThreadCount * 2 + 100);
-    mServer.setThreadPool(threadPool);
+
+    mServer = new Server(threadPool);
+
+    mServerConnector = new ServerConnector(mServer);
+    mServerConnector.setPort(mAddress.getPort());
+    mServerConnector.setHost(mAddress.getAddress().getHostAddress());
+
+    mServer.addConnector(mServerConnector);
```

Comment:
Yeah I guess we would need to deal with the race condition, and it doesn't look like there's an easy way to reserve an ephemeral port.
For this PR, I've updated the code to bind the port in the constructor.